DataFormats/PatCandidates/interface/Jet.h

```diff
@@ -441,6 +441,9 @@ namespace pat {
       ///    Else return the reco Jet number of constituents
       const reco::Candidate * daughter(size_t i) const override;
 
+      reco::CandidatePtr daughterPtr( size_t i ) const;
+      const reco::CompositePtrCandidate::daughters & daughterPtrVector() const;
```

Comment:
I now realize that these methods are not virtual.
So, an overload here will not be visible in a code using pointers/refs upstream in the inheritance.

Compared to that, the numberOfDaughters is virtual and will be visible there and will trigger inconsistencies.

Reply:
Yes, this has been the case since even before my time. (Can I reiterate how much the Candidate model is a pain?)